env0/resource_environment_test.go

```diff
@@ -185,6 +185,72 @@ func TestUnitEnvironmentResource(t *testing.T) {
 			})
 		})
 
+		t.Run("prevent auto deploy", func(t *testing.T) {
+			templateId := "template-id"
+			newTemplateId := "new-template-id"
+			truthyFruity := true
+
+			environment := client.Environment{
+				Id:        uuid.New().String(),
+				Name:      "name",
+				ProjectId: "project-id",
+				LatestDeploymentLog: client.DeploymentLog{
+					BlueprintId: templateId,
+				},
+			}
+
+			testCase := resource.TestCase{
+				Steps: []resource.TestStep{
+					{
+						Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+							"name":                environment.Name,
+							"project_id":          environment.ProjectId,
+							"template_id":         templateId,
+							"force_destroy":       true,
+							"prevent_auto_deploy": true,
+						}),
+						Check: resource.ComposeAggregateTestCheckFunc(
+							resource.TestCheckResourceAttr(accessor, "id", environment.Id),
+							resource.TestCheckResourceAttr(accessor, "name", environment.Name),
+							resource.TestCheckResourceAttr(accessor, "project_id", environment.ProjectId),
+							resource.TestCheckResourceAttr(accessor, "template_id", templateId),
+							resource.TestCheckResourceAttr(accessor, "prevent_auto_deploy", "true"),
+						),
+					},
+					{
+						Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
```

Comment:
this step isn't required here...

Reply:
it fails without it - care to explain what it does?

Reply:
sorry my bad, had to fix the assertions as well - the API of those tests is confusing AF 😅 
[removed unnecessary step](https://github.com/env0/terraform-provider-env0/pull/804/commits/e9a6644e1f1d7019b8a055b7e2ff31c40fe99f22)